### PR TITLE
Configure ECS deployment w/ github actions

### PR DESCRIPTION
**Pull request by KaylaBrady**
[Ticket](https://app.asana.com/0/1199969637453355/1200247733385052)

This PR adds the ECS deployment GHA and adds a Dockerfile to specify how to build a Draft image. Paul has already created a IAM user for Draft deployment.

 I referenced [arrow's Dockerfile](https://github.com/mbta/arrow/blob/master/Dockerfile) in creating this dockerfile. I noticed Skate's `.dockerignore` additionally ignores everything from `priv` that isn't in `!/priv/repo/migrations` -- is that a pattern we'd want to follow as well, or is it standard for everything in `priv` to be included?

### CODE REVIEW

File: .github/workflows/deploy-ecs.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    concurrency: dev
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: mbta/actions/build-push-ecr@v1
+        id: build-push
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          docker-repo: ${{ secrets.DOCKER_REPO }}
```
Comment (KaylaBrady) [inline .github/workflows/deploy-ecs.yml:22]: Are these secrets (+ SLACK_WEBHOOK) already setup for Draft? Can I be added as an admin of this repo, or is that permission restricted?

Comment (paulswartz) [reply .github/workflows/deploy-ecs.yml:22]: SLACK_WEBHOOK is already set up (it's global to all MBTA projects).
I added tje AWS keys earlier, and DOCKER_REPO today.

File: config/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,36 @@
+FROM hexpm/elixir:1.11.4-erlang-23.3-debian-buster-20210208 as builder
```
Comment (paulswartz) [inline config/Dockerfile:1]: I believe the build script is looking for `Dockerfile` to live at the top-level.

File: config/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+ENV LANG=C.UTF-8 \
+  MIX_ENV=prod
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+  ca-certificates curl git
+
+RUN  curl -sL https://deb.nodesource.com/setup_14.x | bash - \
```
Comment (paulswartz) [inline config/Dockerfile:9]: We're trying to migrate applications to install Node as part of a multi-stage build. Ian has an example for the parking repo: https://github.com/mbta/parking_lpr/blob/idw-ecs-deploy/Dockerfile

File: config/.dockerignore [hunk 4]
```diff
@@ -0,0 +1,4 @@
+_build
```
Comment (paulswartz) [inline config/.dockerignore:1]: I believe the build script is looking for this to live at the top-level.

**Review by paulswartz — CHANGES_REQUESTED** [review]

File: config/Dockerfile (hunk 3 above)
Comment (KaylaBrady) [reply config/Dockerfile]: Oh cool, I've updated following that example!

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /root
+EXPOSE 4000
+ENV MIX_ENV=prod TERM=xterm LANG="C.UTF-8" PORT=4000
+
+# add frontend assets with manifests from app build container
+COPY --from=app-builder /root/priv/static ./priv/static
+# add application artifact comipled in app build container
+COPY --from=app-builder /root/_build/prod/rel/screens .
```
Comment (ianwestcott) [inline Dockerfile:51]: I suspect this path is incorrect since it references `screens`

Comment (paulswartz) [reply Dockerfile:51]: 👍 @KaylaBrady you can test out the build locally w/ `docker build .`: it should get to the end successfully.

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+# add frontend assets compiled in node container, required by phx.digest
+COPY --from=assets-builder /root/priv/static ./priv/static
+
+RUN mix do compile --force, phx.digest, release
+
+# finally, use a debian container for the runtime environment
+FROM debian:buster
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+  libssl1.1 libsctp1 curl \
```
[the remark this answers is not in the document]
Comment (KaylaBrady) [reply Dockerfile]: True -- now that curl isn't needed for installing node it can be removed!

File: Dockerfile (hunk 5 above)
Comment (KaylaBrady) [reply Dockerfile]: 🤦‍♀️ Thanks for catching! I've now tested locally with docker build & docker run w/ my local DB configuration for environment variables and confirmed there were no errors logged.

**Review by paulswartz — APPROVED** [review]
🍰

**Pull request merged**